File: 5/5-296_classic.py
# Автор В.Н. Шубинкин

def is_correct(num):
    s = 0
    while num:
        s += num % 10
        num //= 10
        d = 2 * (num % 10)
        d = d % 10 + d // 10
        s += d
        num //= 10
    return s % 10 == 0


num = 1234_5678_9101_1121
num += 1
while not is_correct(num):
    num += 1
print(num % 10**8)

File: Dima/5/5-296_classic.pas
// Автор В.Н. Шубинкин

function is_correct(num: BigInteger): boolean;
begin
  var s: BigInteger;
  while num > 0 do
  begin
    s += num mod 10;
    num := num div 10;
    var d := num mod 10 * 2;
    s += d div 10 + d mod 10;
    num := num div 10
  end;
  result := s mod 10 = 0
end;

begin
  var num := 1234567891011121;
  num += 1;
  while not is_correct(num) do num += 1;
  print(num mod 10bi**8)
end.